core/Tracker/GoalManager.php

```diff
@@ -220,6 +221,22 @@ private function detectNumericGoalMatch($goal, Action $action, VisitProperties $
                 $visitDurationInSecs = $request->getCurrentTimestamp() - ((int) $firstActionTime);
                 $valueToMatchAgainst = $visitDurationInSecs / 60;
                 break;
+            case 'visit_nb_pageviews':
+                $totalInteractions = (int) $visitProperties->getProperty('visit_total_interactions');
+
+                if ($action instanceof ActionPageview) {
+                    $totalInteractions++; // current action won't be included in interaction count
+                }
+
+                if (empty($totalInteractions)) {
+                    return null;
+                }
+
+                // total interactions includes searches, so get number of searches and remove them
+                $totalSearches = (int) $visitProperties->getProperty('visit_total_searches');
```

Comment:
Couldn't find a smarter way to get the number of page views without performing extra queries. Maybe someone else has a better idea?

Reply:
Can you point out the extra query? I can't find it in the changes. (Note: VisitProperties->getProperty() just gets from an array, it doesn't perform a query AFAIK.)

Reply:
That may have been a bit ambiguous. The current solution has no additional queries. However, I have not found any other possibility that would work without additional queries.

Reply:
That makes this solution sound like the best one... unless there's a downside to it? I haven't really thought about it but I don't see anything obviously problematic.

Reply:
@sgiehl can you answer ^? Is there anything that you know about this approach that isn't ideal? Just so I know why you think it's maybe not the best solution, even though it has no additional queries.

Reply:
If we ever decide to count more actions into `visit_total_interactions`, or maybe a plugin manipulates that value to add custom interactions, the calculation might not longer be correct anymore.

Reply:
I see, in that case it sounds like the best solution is a new dimension like `visit_total_pageviews` (if something like that doesn't exist). Since we can't add that until Matomo 5, this seems like the next best solution. We could create an issue to move to a new dimension, or just wait until 5.x to implement this.

cc @tsteur

Reply:
@sgiehl @diosmosis seems the only way to do it for now. Can we leave a comment in `shouldCountInteraction` in visit total interactions that adjusting the logic would break this feature? For example this will happen when we add features like https://github.com/matomo-org/matomo/issues/9199 where we can say a pageview or an event was interactive or non-interactive. This may for example cause some page views to not be counted as interaction or it may cause an event to be counted as interactive.

Personally, I would rather not work on this feature for now as it may cause more problems than it does good and not sure why the issue is marked as major as I don't think it be used that often. @mattab any thoughts if the feature could wait until Matomo 5?

Reply:
Sounds good to me to wait until Matomo 5 and a new dimension/column can be added :+1:

Reply:
I've updated the PR so it uses a new dimension and moved it to 5.0 milestone...